core/src/main/java/com/radixdlt/p2p/addressbook/HighPriorityPeerList.java

```diff
@@ -0,0 +1,126 @@
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.p2p.addressbook;
+
+import static com.radixdlt.lang.Unit.unit;
+
+import com.google.common.base.Preconditions;
+import com.radixdlt.lang.Unit;
+import com.radixdlt.p2p.NodeId;
+import com.radixdlt.utils.LRUCache;
+import java.util.*;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A thread-safe, size-constrained, LRU-ordered list of peers that are considered in some way
+ * important (validators, for example). Implementation note: this is simply a wrapper which adds
+ * synchronization to a non-thread-safe {@link LRUCache}.
+ */
+@ThreadSafe
+public final class HighPriorityPeerList {
+
+  /** Underlying non-synchronized LRU cache. */
+  private final LRUCache<NodeId, Unit> nodeIds;
+
+  /**
+   * Creates a list with the given maximum size and initial contents. Note: the initial contents may
+   * get truncated (from the beginning) if larger than maximum size.
+   */
+  public HighPriorityPeerList(int maxSize, List<NodeId> initialNodeIds) {
+    this.nodeIds = new LRUCache<>(maxSize);
+    for (var nodeId : initialNodeIds) {
+      Preconditions.checkNotNull(nodeId);
```

Comment:
_(note from author)_

I'd be very surprised if the `null`s came from persistence; yet I figured this check won't harm us (and it will narrow down the analysis if NPEs happen again).